process/block/shardblockProposal.go

```diff
@@ -230,6 +230,9 @@ func (sp *shardProcessor) selectIncomingMiniBlocks(
 	orderedMetaBlocksHashes [][]byte,
 	haveTime func() bool,
 ) error {
+
+	sp.gasComputation.Reset()
```

Comment:
can we move this instead on the CreateBlockProposal instead before any selection (either for miniblocks or transactions)
e.g I would move it next to `sp.miniBlocksSelectionSession.ResetSelectionSession()`